### PR TITLE
Throw if trying to convert to or from flexible synced Realms

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

See https://github.com/realm/realm-core/issues/5798

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [x] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: src/js_realm.hpp [hunk 1]
```diff
@@ -1607,6 +1607,12 @@ realm::Realm::Config RealmClass<T>::write_copy_to_helper(ContextType ctx, Object
     ObjectDefaultsMap defaults;
     ConstructorMap constructors;
     bool schema_updated = get_realm_config(ctx, args.count, args.value, config, defaults, constructors);
+
+    // TODO: remove when https://github.com/realm/realm-core/issues/5798 is fixed
+    if (config.sync_config && config.sync_config->flx_sync_requested) {
+        throw std::runtime_error("`writeCopyTo` does not support flexible sync");
```
Comment (tomduncalf) [inline src/js_realm.hpp:1613]: We could maybe word this a little more positively as I suspect this is quite frustrating to hit as a user... not sure if it's needed though
```suggestion
        throw std::runtime_error("`writeCopyTo` does not currently support flexible sync");
```

**Review by tomduncalf — APPROVED** [review]
This well prevent a lot of confusion 👍

File: src/js_realm.hpp (hunk 1 above)
Comment (kneth) [reply src/js_realm.hpp]: I have updated the message - also in the test.

**Comment by kneth** [comment]
Fixed in Realm Core v12.7.0.

**Pull request closed without merging**